app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawer.kt

```diff
@@ -0,0 +1,237 @@
+                FahrplanFragment.context, R.color.event_drawable_selection_stroke)
+        eventDrawableRippleColor = ContextCompat.getColor(
+                FahrplanFragment.context, R.color.event_drawable_ripple)
+        trackNameBackgroundColorDefaultPairs = TrackBackgrounds.getTrackNameBackgroundColorDefaultPairs(context)
+        trackNameBackgroundColorHighlightPairs = TrackBackgrounds.getTrackNameBackgroundColorHighlightPairs(context)
+    }
+
+    private fun updateEventView(eventView: View, lecture: Lecture) {
+        val bell = eventView.findViewById<ImageView>(R.id.bell)
+        bell.visibility = if (lecture.hasAlarm) View.VISIBLE else View.GONE
+        var title = eventView.findViewById<TextView>(R.id.event_title)
+        title.typeface = boldCondensed
+        title.text = lecture.title
+        title = eventView.findViewById(R.id.event_subtitle)
+        title.text = lecture.subtitle
+        title = eventView.findViewById(R.id.event_speakers)
+        title.text = lecture.formattedSpeakers
+        title = eventView.findViewById(R.id.event_track)
+        title.text = lecture.formattedTrackText
+        title.contentDescription = lecture.getFormattedTrackContentDescription(eventView.context)
+        val recordingOptOut = eventView.findViewById<View>(R.id.novideo)
+        if (recordingOptOut != null) {
+            recordingOptOut.visibility = if (lecture.recordingOptOut) View.VISIBLE else View.GONE
+        }
+        setLectureBackground(lecture, eventView)
+        setLectureTextColor(lecture, eventView)
+        eventView.setOnClickListener(fahrplanFragment)
+        eventView.setOnCreateContextMenuListener(fahrplanFragment)
+        eventView.isLongClickable = true
+        eventView.tag = lecture
+    }
+
+    fun createLectureViews(room: LinearLayout, lectureLayoutParams: Map<Lecture, LinearLayout.LayoutParams>, standardHeight: Int) {
+        room.removeAllViews()
+
+        for (entry in lectureLayoutParams) {
+            val lecture = entry.key
+            val params = entry.value
+
+            val event = inflater.inflate(R.layout.event_layout, null)
+            val height = standardHeight * (lecture.duration / 5)
+            room.addView(event, LinearLayout.LayoutParams.MATCH_PARENT, height)
+
+            val lp = event.layoutParams as LinearLayout.LayoutParams
+            lp.bottomMargin = params.bottomMargin
+            lp.topMargin = params.topMargin
+            event.layoutParams = lp
+
+            updateEventView(event, lecture)
+        }
+    }
+
+    fun setLectureBackground(event: Lecture, eventView: View) {
+        val context = eventView.context
+        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
+        val defaultValue = resources.getBoolean(R.bool.preferences_alternative_highlight_enabled_default_value)
+        val alternativeHighlightingIsEnabled = prefs.getBoolean(
+                BundleKeys.PREFS_ALTERNATIVE_HIGHLIGHT, defaultValue)
```

Comment:
See https://github.com/EventFahrplan/EventFahrplan/pull/256#discussion_r404126003, and I did not plan to unit test this method ...

Reply:
I thought so. However, since the old code part was architected without testing in mind I encourage you to promote a testable new class without any pressure to this right now yourself. ([boy scouting rule](https://medium.com/@biratkirat/step-8-the-boy-scout-rule-robert-c-martin-uncle-bob-9ac839778385))

Reply:
I have decided to leave it as is for now.

Reply:
Accepted as is.